sawtooth-sdk-transaction-processor/src/main/java/sawtooth/sdk/processor/StreamContext.java

```diff
@@ -148,14 +156,110 @@ public ZmqContext(final Stream myStream, final String myContextId) {
     ArrayList<String> addressesThatWereSet = new ArrayList<String>();
     if (setResponse != null) {
       if (setResponse.getStatus() == TpStateSetResponse.Status.AUTHORIZATION_ERROR) {
-  throw new InvalidTransactionException("Tried to set unauthorized address " + addressValuePairs.toString());
+        throw new InvalidTransactionException("Tried to set unauthorized address " + addressValuePairs.toString());
       }
       for (String address : setResponse.getAddressesList()) {
-  addressesThatWereSet.add(address);
+        addressesThatWereSet.add(address);
       }
     }
 
     return addressesThatWereSet;
   }
 
+  @Override
+  public final Collection<String> deleteState(final Collection<String> addresses)
+      throws InternalError, InvalidTransactionException {
+    TpStateDeleteRequest delRequest = TpStateDeleteRequest.newBuilder().addAllAddresses(addresses)
+        .setContextId(this.contextId).build();
+    Future future = stream.send(Message.MessageType.TP_STATE_DELETE_REQUEST, delRequest.toByteString());
+    TpStateDeleteResponse delResponse = null;
+    try {
+      delResponse = TpStateDeleteResponse.parseFrom(future.getResult(TIME_OUT));
+    } catch (InterruptedException iee) {
+      throw new InternalError(iee.toString());
+    } catch (InvalidProtocolBufferException ipbe) {
+      // server didn't respond with a DeleteResponse
+      throw new InternalError(ipbe.toString());
+    } catch (ValidatorConnectionError vce) {
+      throw new InternalError(vce.toString());
+    } catch (Exception e) {
+      throw new InternalError(e.toString());
+    }
+    List<String> results = new ArrayList<>();
+    if (delResponse != null) {
+      if (delResponse.getStatus() == TpStateDeleteResponse.Status.AUTHORIZATION_ERROR) {
+        throw new InvalidTransactionException("Tried to delete unauthorized address " + addresses.toString());
+      }
+      for (String entry : delResponse.getAddressesList()) {
+        results.add(entry);
+      }
+    }
+    if (results.isEmpty()) {
+      throw new InternalError("State Error, no addresses found for delete request:" + addresses.toString());
```

Comment:
This is interesting. I looked at the python and rust SDKs and they don't trap for empty lists and then raise an internal error. I think the simplest thing to do would be follow that pattern (ignore the possibility of an empty address list).
If the interface contract is the protobuf, a repeated field may have 0 elements. 
I thought about whether this is more robust, but I think there's a chance that if you sent an empty list in the request than it will cause and endless loop of internal errors (because the empty list will be returned in the response). And then you'd have to trap for that kind of request which the other SDKs don't do.

Reply:
So, I drew that from the existing getState implementation - reasoning that if a Get that returns nothing is an error, then a  Delete which deletes nothing is also probably an error.  But to your point the other SDK's don't appear to throw an error on an empty Get response either.

Reply:
So per our conversation, going to remove that check on getState and deleteState since it doesn't exist in the other SDK's at this time.